### PR TITLE
[Merged by Bors] - feat: maximal upper/minimal lower bounds of finset elements

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
Originally motivated by the Carleson project. We rename `Finite.exists_ge_minimal` to `Finite.exists_minimal_le` and add `Finset` and `Set.Finite` versions.

### CODE REVIEW

File: Mathlib/Data/Finset/Minimal.lean [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/-- Every element in a finset over a partially ordered type has a maximal upper bound. -/
+lemma exists_maximal_upper_bound (h : a ∈ s) : ∃ m ≥ a, Maximal (· ∈ s) m := by
+  classical let C : Finset α := s.filter (a ≤ ·)
+  obtain ⟨m, hm, maxm⟩ := C.exists_maximal ⟨a, by simp [C, h]⟩
+  simp_rw [C, mem_filter] at hm maxm
+  use m, hm.2; rw [maximal_iff]
+  exact ⟨hm.1, fun b mb lb ↦ eq_of_le_of_not_lt lb (maxm b ⟨mb, hm.2.trans lb⟩)⟩
+
+/-- Every element in a finset over a partially ordered type has a minimal lower bound. -/
+lemma exists_minimal_lower_bound (h : a ∈ s) : ∃ m ≤ a, Minimal (· ∈ s) m := by
```
Comment (urkud) [inline Mathlib/Data/Finset/Minimal.lean:31]: Can you use `OrderDual` here?

File: Mathlib/Data/Finset/Minimal.lean [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* `Finset.exists_maximal_upper_bound` says that it has a maximal upper bound.
+* `Finset.exists_minimal_lower_bound` says that it has a minimal lower bound.
+-/
+
+namespace Finset
+
+variable {α : Type*} [PartialOrder α] {s : Finset α} {a : α}
+
+/-- Every element in a finset over a partially ordered type has a maximal upper bound. -/
+lemma exists_maximal_upper_bound (h : a ∈ s) : ∃ m ≥ a, Maximal (· ∈ s) m := by
```
Comment (urkud) [inline Mathlib/Data/Finset/Minimal.lean:23]: See also [Finite.exists_ge_minimal](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Order.html#Finite.exists_ge_minimal), which should be renamed to `Finite.exists_le_minimal`. Also, please add a version for `Set.Finite`.

**Comment by YaelDillies** [comment]
As Yury is pointing out, these lemmas already exist. The names are a bit all over the place, though, so one possible action would be to find a nice naming scheme and make mathlib follow it.

File: Mathlib/Data/Finset/Minimal.lean (hunk 2 above)
Comment (Parcly-Taxel) [reply Mathlib/Data/Finset/Minimal.lean]: Done.

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**